waspc/src/Analyzer/Parser/AST.hs

```diff
@@ -0,0 +1,26 @@
+module Analyzer.Parser.AST where
```

Comment:
A thought: AST here is submodule of Parser. But, Decl is not submodule of Evaluator. That is inconsistency.

It makes sense to me to keep AST, TypedAST and Decl all at top lvl, since each of them is used in two parts of Analyzer. But we could also keep them as Parser.AST, TypeChecker.TypedAST and Evaluator.Decl, which feels ok because they are the results of those parts of analyzer. I find both approaches reasonable. if I had to choose one, I  like the second one a bit better.

Reply:
Aha, well that is a good argument for keeping it top level!
Well, I think at the end it is really hard to say what is better here, it also feels very similar! I am still inclined to keep it all top level or all in the corresponding modules, instead of keeping some top level (Decl) and some not (others). So I would probably put it into Evaluator if it was up to me. But I don't have a strong argument really nor do I think it is very important, so I am fine with however you do it -> do as you prefer and all good!

Reply:
I'll leave it at the top level.